OutOfSchool/OutOfSchool.Common/Extensions/StringExtensions.cs

```diff
@@ -35,4 +36,15 @@ public static string Right(this string value, int length)
             ? value[^length..]
             : value;
     }
+
+    public static bool ContainsCharacterType(this string source, Func<char, bool> characterType)
+    {
+        return source.Any(characterType);
+    }
+
+    public static bool ContainsAnySymbol(this string source, string symbols)
+    {
+        return !string.IsNullOrEmpty(source) && !string.IsNullOrEmpty(symbols)
+            && symbols.Any(source.Contains);
+    }
```

Comment:
(Small side note) This is not a call to action, just for your information.
I've tried to measure performance (on .Net 8.0) using different search approaches - regExp, your code, and dumb single-iteration algorithm. Code and results - on screenshot below
![image](https://github.com/ita-social-projects/OoS-Backend/assets/85107137/cd44ff0f-b32b-406f-b178-d26ed68dbc54)

Reply:
@VadymLevkovskyi Thank you! It is very useful and interesting for me. Before I saw your benchmarks, I rewrote `IsValidPassword` because I thought that it was necessary to limit the characters in the password to only permitted requirements:
![image](https://github.com/ita-social-projects/OoS-Backend/assets/107883170/b5b0dae4-7300-4734-be7b-23af36990ead)

Reply:
And your variant for checking char "type" is even faster than `CharUnicodeInfo.GetUnicodeCategory(c)` I've used (for the code see my previous scren). 

Your code with early exit approach and performance comparison is below
![image](https://github.com/ita-social-projects/OoS-Backend/assets/85107137/3f3cca02-98c4-44df-bc59-0c0c49cb88c4)